airflow/models/dag.py

```diff
@@ -1468,14 +1472,28 @@ def create_dagrun(self,
         :param session: database session
         :type session: sqlalchemy.orm.session.Session
         """
+        if run_id:
```

Comment:
```suggestion
        if run_id and not run_type:
```

We should be able to specify a type and an run_id, don't you think? (Right now this would blindly overwrite the provided run_type if run_id is specified.)

Reply:
Missing this change

Reply:
Will add it once we agree on index. I want to avoid another multiple rebases 😉

Reply:
Done